packages/zwave-js/src/lib/commandclass/EntryControlCC.ts

```diff
@@ -513,9 +438,35 @@ export class EntryControlCCEventSupportedReport extends EntryControlCC {
 
 		validatePayload(this.payload.length >= offset + 4);
 		this.minKeyCacheSize = this.payload[offset];
+		validatePayload(
+			this.minKeyCacheSize >= 1 && this.minKeyCacheSize <= 32,
+		);
 		this.maxKeyCacheSize = this.payload[offset + 1];
+		validatePayload(
+			this.maxKeyCacheSize >= this.minKeyCacheSize &&
+				this.maxKeyCacheSize <= 32,
+		);
 		this.minKeyCacheTimeout = this.payload[offset + 2];
 		this.maxKeyCacheTimeout = this.payload[offset + 3];
+
+		const keyCacheSizeValueId = getKeyCacheSizeStateValueID(
+			this.endpointIndex,
+		);
+		this.getValueDB().setMetadata(keyCacheSizeValueId, {
+			...ValueMetadata.UInt8,
+			min: this.minKeyCacheSize,
+			max: this.maxKeyCacheSize,
+		});
+
+		const keyCacheTimeoutValueId = getKeyCacheTimeoutStateValueID(
+			this.endpointIndex,
+		);
+		this.getValueDB().setMetadata(keyCacheTimeoutValueId, {
+			...ValueMetadata.UInt8,
+			min: this.minKeyCacheTimeout,
+			max: this.maxKeyCacheTimeout,
+		});
+
```

Comment:
Looks cleaner now - please remove the `@ccValue({ internal: true })` annotations below this line for the properties that are now part of metadata.